### ISSUE TITLE
false "Attempted to reuse member name in Enum definition"

### ISSUE THREAD

**Issue by belm0** [python/mypy#16876]
**Bug Report**

When using an `if`/`else` in class scope of an Enum definition, there is a false error "Attempted to reuse member name in Enum definition".

I understand mypy might have an opinion about this pattern, but I noticed there is no equivalent issue for a plain class.  So this seems to be an issue specific to Enum that could potentially be relaxed.

**To Reproduce**

```python
class Foo(Enum):
    if IS_BAZ:
        DISABLED = 16
    else:
        # error: Attempted to reuse member name "DISABLED" in Enum definition "Foo"  [misc]
        DISABLED = 256

class Bar:
    if IS_BAZ:
        DISABLED = 16
    else:
        DISABLED = 256   # (no error)
```

**Expected Behavior**

no mypy errors

**Actual Behavior**

```
error: Attempted to reuse member name "DISABLED" in Enum definition "Foo"  [misc]
```

**Your Environment**

- Mypy version used:  1.8.0
- Python version used:  3.8
